File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/PowerRemovedMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import { AbstractPower } from "../Power/AbstractPower";
import AbstractCreature from "../Unit/AbstractCreature";

export class PowerRemovedMessage implements IMessage{
    public static readonly Id = "PowerRemovedMessage";
    public constructor(public Power:AbstractPower,public Owner:AbstractCreature) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/PowerRemovedMessage.cs
using ModTheSpire.Power;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///能力移出
    public class PowerRemovedMessage : IEventHandle
    {
        public AbstractPower Power;
        public AbstractCreature Unit;

        public PowerRemovedMessage(AbstractPower power, AbstractCreature unit)
        {
            Power = power;
            Unit = unit;
        }
    }
}